File: as_ios/Astra Staging Portal/Astra Staging Portal/MenuItem.swift
//
//  MenuItem.swift
//  Astra Staging Portal
//
//  All possible menu destinations. Order in `defaultOrder` is the seed order
//  for new users; once they customize, `MenuPreferences` persists their order.
//

import Foundation

enum MenuItem: String, CaseIterable, Identifiable, Codable {
    case tasks
    case items
    case me
    case consultation
    case sales
    case hr

    var id: String { rawValue }

    var label: String {
        switch self {
        case .tasks:        return "Tasks"
        case .items:        return "Items"
        case .me:           return "Me"
        case .consultation: return "Consultation"
        case .sales:        return "Sales"
        case .hr:           return "HR"
        }
    }

    var sfSymbol: String {
        switch self {
        case .tasks:        return "checklist"
        case .items:        return "shippingbox"
        case .me:           return "person.circle"
        case .consultation: return "person.2.wave.2"
        case .sales:        return "briefcase"
        case .hr:           return "building.2"
        }
    }

    /// Min user.roleLevel required to see this item (1=execution, 2=manager, 3=owner).
    var minRoleLevel: Int {
        switch self {
        case .tasks, .items, .me, .consultation: return 1
        case .sales: return 2
        case .hr:    return 3
        }
    }

    static let defaultOrder: [MenuItem] = [.tasks, .items, .consultation, .me, .sales, .hr]
}

File: as_android/AstraStagingPortal/app/src/main/java/com/astrastaging/portal/ui/main/MenuItem.kt
package com.astrastaging.portal.ui.main

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.Apartment
import androidx.compose.material.icons.outlined.Business
import androidx.compose.material.icons.outlined.CheckCircle
import androidx.compose.material.icons.outlined.Inventory2
import androidx.compose.material.icons.outlined.MoreHoriz
import androidx.compose.material.icons.outlined.Person
import androidx.compose.material.icons.outlined.SupportAgent
import androidx.compose.ui.graphics.vector.ImageVector

/**
 * All possible menu destinations. The order in [defaultOrder] is the seed
 * for new installs; once the user customizes, MenuPreferencesStore persists
 * their preferred order.
 *
 * minRoleLevel: 1 = execution (mover/stager), 2 = manager, 3 = owner.
 */
enum class MenuItem(
    val key: String,
    val label: String,
    val icon: ImageVector,
    val minRoleLevel: Int,
) {
    TASKS("tasks", "Tasks", Icons.Outlined.CheckCircle, 1),
    ITEMS("items", "Items", Icons.Outlined.Inventory2, 1),
    ME("me", "Me", Icons.Outlined.Person, 1),
    CONSULTATION("consultation", "Consultation", Icons.Outlined.SupportAgent, 1),
    SALES("sales", "Sales", Icons.Outlined.Business, 2),
    HR("hr", "HR", Icons.Outlined.Apartment, 3);

    companion object {
        val defaultOrder: List<MenuItem> = listOf(TASKS, ITEMS, CONSULTATION, ME, SALES, HR)
        fun fromKey(key: String): MenuItem? = entries.firstOrNull { it.key == key }
        val MoreIcon: ImageVector get() = Icons.Outlined.MoreHoriz
    }
}
